src/plugin.ts

```diff
@@ -40,6 +42,21 @@ export class PluginTypeEnvironment extends TypeEnvironment {
         return this.getType(this.checker.lookupTypeAt(n, this.sinapSourceFile));
     }
 
+    private getFunctionSignatures(name: string, node: ts.Node) {
+        const functionSymbol = this.checker.getSymbolsInScope(node, ts.SymbolFlags.Function)
+            .filter((a) => a.name === name)[0];
+        if (functionSymbol === undefined) {
+            throw new Error(`function "${name}" not found`);
+        }
+        const functionType = this.checker.getTypeOfSymbol(functionSymbol);
+        const sig = functionType.getCallSignatures();
+        return sig.map(s => 
+            [
+                s.getParameters().map(p => this.getType(this.checker.getTypeOfSymbol(p))),
```

Comment:
This lambda should really be multiple lines and have the parameters assigned with good names for readability.